src/data/IFormState.ts

```diff
@@ -0,0 +1,24 @@
+// The FormStatus is to represent FormInput status
+export interface IFormState {
```

Comment:
How does a field know its errors? Should this type get passed back down as a prop to the FormInput?